data_steward/retraction/create_deid_map.py

```diff
@@ -0,0 +1,172 @@
+    :return: list of combined_datasets that should contain a _deid_map table
+    """
+    all_datasets_obj = utils.bq.list_datasets(project_id)
+    all_datasets = [d.dataset_id for d in all_datasets_obj]
+    deid_datasets = []
+
+    # Keep only deid datasets with standard cdr release naming convention
+    release_datasets = [
+        d for d in all_datasets if consts.CURRENT_RELEASE_REGEX.match(d)
+    ]
+    # Filter for deid datasets
+    for dataset in release_datasets:
+        if bool(re.match(DEID_REGEX, dataset)) is True:
+            deid_datasets.append(dataset)
+    # Find corresponding combined dataset for each deid dataset
+    deid_and_combined_df = get_corresponding_combined_dataset(
+        all_datasets, deid_datasets)
+    return deid_and_combined_df
+
+
+def get_corresponding_combined_dataset(all_datasets, deid_datasets):
+    """
+    Takes in all datasets in a project and a list of all the deid datasets in that project and returns the corresponding
+    combined dataset to that deid dataset
+    :param all_datasets: list of all datasets in the given bq project
+    :param deid_datasets: list of all deid datasets in the given bq project
+    :return: list of combine datasets
+    """
+    deid_and_combined_datasets_df = pd.DataFrame(
+        columns=['deid_dataset', 'combined_dataset'])
+    # Get release from deid dataset for both older and current release tags and search for combined dataset
+    for d in deid_datasets:
+        release = d.split('_')[0]
+        prefix = release[1:]
+        combined = prefix + '_combined'
+
+        if combined in all_datasets:
+            new_row = pd.Series({
+                'deid_dataset': d,
+                'combined_dataset': combined
+            })
+            deid_and_combined_datasets_df = deid_and_combined_datasets_df.append(
+                new_row, ignore_index=True)
+
+    return deid_and_combined_datasets_df
+
+
+def check_if_deid_map_exists(project_id, dataset):
+    """
+    Checks if _deid_map table exists in the given dataset.
+    :param project_id: bq name of project_id
+    :param dataset: bq name of dataset
+    :return: returns True, False or 'rename required' which had 'deid_map' table instead of '_deid_map'
+    """
+    table_info_df = utils.bq.get_table_info_for_dataset(project_id, dataset)
+    column_list = table_info_df['table_name'].tolist()
+
+    if '_deid_map' in column_list:
+        return consts.SKIP
+    elif 'deid_map' in column_list:
+        return consts.RENAME
+    else:
+        return consts.CREATE
+
+
+def rename_deid_map_table_query(project_id, dataset):
+    return consts.RENAME_DEID_MAP_TABLE_QUERY.format(project=project_id,
+                                                     dataset=dataset)
+
+
+def create_deid_map_table_query(project, dataset):
+    return consts.CREATE_DEID_MAP_TABLE_QUERY.format(project=project,
+                                                     dataset=dataset)
+
```

Comment:
These functions and their tests can be removed since they're not really doing anything much

Reply:
I'm using these functions in `create_deid_map_table_queries` should I just move the logic down there and remove the functions?

Reply:
Yeah that should suffice